### DISCUSSION TITLE
Empirical Robustness Metric Support for all Evasion attacks

### DISCUSSION THREAD

**Discussion by shreyakhandelwal07** [Trusted-AI/adversarial-robustness-toolbox#1228] [Q&A] [answered]
Hi, I can see that the attack methods currently supported for evaluating empirical robustness are Fast Gradient Sign Method and Hop Skip Jump. 
https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/3d86b8d66b646fd736beda570254786bcff55e0e/art/metrics/metrics.py#L97
But in theory, it can be evaluated for all evasion attacks, right? 

Are there any plans to support more attack methods for evaluating empirical robustness?

**Comment by beat-buesser** [comment, marked as answer]
Hi @shreyakhandelwal07 Thank you very much for your question! I have moved your issue to our new Discussion forum.

Yes, `empirical_robustness` should be applicable to all evasion attacks. The `SUPPORTED_METHODS` is not exclusive and would need to be adapted by the user. A few drawbacks and reason for caution are that the implementation of `empirical_robustness` includes the default values for the attacks in `SUPPORTED_METHODS` which might need adaptations to a specific problem, it does not adapt to models and defences, and it attacks the predictions of the model instead of the stronger attack against true labels. I think these are the main reasons why it has not been updated recently.

Would you be interested to update and extend `empirical_robustness`?
